UKHO.MaritimeSafetyInformation.Web/UKHO.MaritimeSafetyInformation.Web/Services/FileShareService.cs

```diff
@@ -72,16 +73,53 @@ public async Task<Stream> FSSDownloadFileAsync(string batchId, string fileName,
                 _logger.LogInformation(EventIds.FSSGetSingleWeeklyNMFileStarted.ToEventId(), "Maritime safety information request for FSS to get single weekly NM file started for batchId:{batchId} and fileName:{fileName} with _X-Correlation-ID:{correlationId}", batchId, fileName, correlationId);
 
                 FileShareApiClient fileShareApi = new(_httpClientFactory, _fileShareServiceConfig.Value.BaseUrl, accessToken);
-                Stream  stream = await fileShareApi.DownloadFileAsync(batchId, fileName);
-              
+                Stream stream = await fileShareApi.DownloadFileAsync(batchId, fileName);
+
                 _logger.LogInformation(EventIds.FSSGetSingleWeeklyNMFileCompleted.ToEventId(), "Maritime safety information request for FSS to get single weekly NM file completed for batchId:{batchId} and fileName:{fileName} with _X-Correlation-ID:{correlationId}", batchId, fileName, correlationId);
-                return stream;  
+                return stream;
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(EventIds.FSSGetSingleWeeklyNMFileResponseFailed.ToEventId(), "Failed to get single weekly NM file from FSS for batchId:{batchId} and fileName:{fileName} with exception:{exceptionMessage} for _X-Correlation-ID:{CorrelationId}", batchId, fileName, ex.Message, correlationId);
+                throw;
+            }
+        }
+
+        public async Task<Stream> FSSDownloadZipFile(string batchId, string fileName, string accessToken, string correlationId)
```

Comment:
To be consistent with other methods:
```suggestion
        public async Task<Stream> FSSDownloadZipFileAsync(string batchId, string fileName, string accessToken, string correlationId)
```